src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -319,10 +325,13 @@ protected void AddCoins(TransactionBuildContext context)
                     UnspentOutputReference item = availableHashList[outPoint];
 
                     coins.Add(new Coin(item.Transaction.Id, (uint)item.Transaction.Index, item.Transaction.Amount, item.Transaction.ScriptPubKey));
+                    sum += item.Transaction.Amount;
                 }
             }
 
-            foreach (UnspentOutputReference item in context.UnspentOutputs)
+            foreach (UnspentOutputReference item in context.UnspentOutputs
+                .OrderByDescending(a => a.Confirmations > 0)
```

Comment:
As pointed out by @MithrilMan this line is probably  bug that was brought over form before
```
 .OrderByDescending(a => a.Confirmations > 0)
```
Should this not change to
```
 .OrderByDescending(a => a.Confirmations)
```

Reply:
At this stage we have just reverted the change entirely, it was causing some pretty insane performance issues, but I agree that makes sense. Can PR that to master separately?

Reply:
Yeah, @codingupastorm rather do in new PR.